### PR TITLE
Added checking for a leftover socket from the previous runs

### PR DESCRIPTION
**Pull request by makhov**
**Issue**
When k0s crashes or gets SIGKILL, the status server socket is not unlinked. This can cause troubles on the next runs.
```
Dec 16 12:31:13 ip-172-31-28-129 k0s[4636]: Error: listen unix /run/k0s/status.sock: bind: address already in use
Dec 16 12:31:13 ip-172-31-28-129 k0s[4636]: 2021/12/16 12:31:13 listen unix /run/k0s/status.sock: bind: address already in use
Dec 16 12:31:13 ip-172-31-28-129 systemd[1]: k0scontroller.service: Main process exited, code=exited, status=1/FAILURE
Dec 16 12:31:13 ip-172-31-28-129 systemd[1]: k0scontroller.service: Failed with result 'exit-code'.
Dec 16 12:33:13 ip-172-31-28-129 systemd[1]: k0scontroller.service: Scheduled restart job, restart counter is at 4.
Dec 16 12:33:13 ip-172-31-28-129 systemd[1]: Stopped k0s - Zero Friction Kubernetes.
```

**What this PR Includes**
In the change, we check if there is a leftover socket and if something is listening on this socket. If not, we remove socket file.

### CODE REVIEW

File: pkg/component/status/status.go [hunk 1]
```diff
@@ -47,6 +55,21 @@ func (s *Status) Init() error {
 	return nil
 }
 
+// checkLeftoverSocket checks if there is a leftover socket from the previous runs. This usually happens on crash or SIGKILL
+func checkLeftoverSocket(socket string) error {
+	_, err := net.Dial("unix", "socket")
```
Comment (ncopa) [inline pkg/component/status/status.go:60]: Shouldn't this be the variable `socket` instead of string literal `"socket"`?

**Comment by ncopa** [comment]
I wonder if we really need to test for `ErrNotExist` and "connection refused" errors?

Wouldn't it be enough to always try to delete it unless we get a successful connection? If there are other errors (like permission denied) we'd fail to delete it, right?

I also think the error message `bind: address already in use` explains better what is going on than a `"socket is busy"`, so I think we can just ignore successful connection and let it fail during bind.

I believe it could be simplified to something like:
```go
	// try remove leftover sockets that nothing is listening on
	_, err := net.Dial("unix", s.Socket)
	if err != nil {
		_ := os.Remove(s.Socket)
	}
```

File: pkg/component/status/status.go (hunk 1 above)
Comment (makhov) [reply pkg/component/status/status.go]: Yes, of course! Thank you!

**Comment by makhov** [comment]
@ncopa Yeah, makes a lot of sense. Simplified it as you proposed. Thanks!

**Review by ncopa — APPROVED** [review]

**Pull request merged**